be/src/vec/olap/vcollect_iterator.cpp

```diff
@@ -34,14 +34,19 @@ VCollectIterator::~VCollectIterator() {}
         }                                                                               \
     } while (false)
 
-void VCollectIterator::init(TabletReader* reader) {
+void VCollectIterator::init(TabletReader* reader, bool force_merge, bool is_reverse) {
     _reader = reader;
     // when aggregate is enabled or key_type is DUP_KEYS, we don't merge
     // multiple data to aggregate for better performance
     if (_reader->_reader_type == READER_QUERY &&
         (_reader->_direct_mode || _reader->_tablet->keys_type() == KeysType::DUP_KEYS)) {
         _merge = false;
     }
+
+    if (force_merge) {
+        _merge = true;
+    }
```

Comment:
_merge = force_merge

Reply:
can not just set _merge = force_merge, since when force_merge == false there is a chance _merge is already set to true before this statement.